packages/react-native-sessions/src/contexts/SessionContext.tsx

```diff
@@ -0,0 +1,390 @@
+                return {
+                  id: account.walletAccountId,
+                  curve: account.curve,
+                  pathFormat: account.pathFormat,
+                  path: account.path,
+                  addressFormat: account.addressFormat,
+                  address: account.address,
+                  createdAt: account.createdAt,
+                  updatedAt: account.updatedAt,
+                };
+              }),
+            };
+          }),
+        );
+
+        const user = userResponse.user;
+
+        setUser({
+          id: user.userId,
+          userName: user.userName,
+          email: user.userEmail,
+          phoneNumber: user.userPhoneNumber,
+          organizationId: whoami.organizationId,
+          wallets,
+        });
+      }
+    }
+  };
+
+  /**
+   * Refreshes the user state.
+   *
+   * - Calls `fetchUser` to update user data.
+   * - Should be run when user data changes.
+   */
+  const refreshUser = async () => {
+    await fetchUser();
+  };
+
+  /**
+   * Clears any scheduled expiration timeouts
+   */
+  const clearTimeouts = () => {
+    if (expiryTimeoutRef.current) clearTimeout(expiryTimeoutRef.current);
+  };
+
+  /**
+   * Schedules session expiration callback
+   * @param expiryTime - The Unix timestamp of session expiration
+   */
+  const scheduleSessionExpiration = (expiryTime: number) => {
+    clearTimeouts();
+    const timeUntilExpiry = expiryTime - Date.now();
+
+    if (timeUntilExpiry > 0) {
+      expiryTimeoutRef.current = setTimeout(() => {
+        clearSession();
+        config.onSessionExpired?.();
+      }, timeUntilExpiry);
+    } else {
+      clearSession();
+      config.onSessionExpired?.();
+    }
+  };
+
+  /**
+   * Retrieves the stored embedded key from secure storage.
+   * Optionally deletes the key from storage after retrieval.
+   *
+   * @param deleteKey Whether to remove the embedded key after retrieval. Defaults to `true`.
+   * @returns The embedded private key if found, otherwise `null`.
+   * @throws If retrieving or deleting the key fails.
+   */
+  const getEmbeddedKey = async (deleteKey = true) => {
+    const credentials = await Keychain.getGenericPassword({
+      service: "turnkey-embedded-key",
+    });
+    if (credentials) {
+      if (deleteKey) {
+        await Keychain.resetGenericPassword({
+          service: "turnkey-embedded-key",
+        });
+      }
+      return credentials.password;
+    }
+    return null;
+  };
+
+  /**
+   * Saves an embedded key securely in storage.
+   *
+   * @param key The private key to store securely.
+   * @throws If saving the key fails.
+   */
+  const saveEmbeddedKey = async (key: string) => {
+    try {
+      await Keychain.setGenericPassword("turnkey-embedded-key", key, {
+        accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+        service: "turnkey-embedded-key",
+      });
+    } catch (error) {
+      throw new Error("Could not save the embedded key.");
```

Comment:
Not a hard requirement but maybe we could define an Error type to give more context about where it is originating? 
for instance we have a [TurnkeyRequestError](https://github.com/tkhq/sdk/blob/20042a3e6c421b4194ecdbbebdde539acfc89916/packages/sdk-browser/src/__types__/base.ts#L32-L49) it might make sense to have like a session error or turnkey react native error? 🤔 
here's another simpler [custom error example](https://github.com/tkhq/sdk/blob/main/packages/wallet-stamper/src/errors.ts) for the wallet stamper